app/models/mixins/aggregation_mixin.rb

```diff
@@ -56,12 +56,9 @@ def all_storages
   end
 
   def aggregate_hardware(from, field, targets = nil)
-    from      = from.to_s.singularize
-    select    = field == :aggregate_cpu_speed ? "cpu_total_cores, cpu_speed" : field
-    targets ||= send("all_#{from}_ids")
-    targets   = targets.collect(&:id) unless targets.first.kind_of?(Integer)
-    hdws      = Hardware.where("#{from}_id" => targets).select(select)
-
-    hdws.inject(0) { |t, hdw| t + hdw.send(field).to_i }
+    from_singular = from.to_s.singularize
+    targets ||= send("all_#{from}")
```

Comment:
this allows for a sub-sql `IN` clause instead of pulling back all the ids to send back to the server.

one of these days I'd like to get rid of the `all_` aliases, but reducing queries is my main concern